python/pywatchman/__init__.py

```diff
@@ -114,16 +126,128 @@ def __init__(self):
 
     LocalFree = ctypes.windll.kernel32.LocalFree
 
-    GetOverlappedResultEx = ctypes.windll.kernel32.GetOverlappedResultEx
-    GetOverlappedResultEx.argtypes = [wintypes.HANDLE,
-                                      ctypes.POINTER(OVERLAPPED), LPDWORD,
-                                      wintypes.DWORD, wintypes.BOOL]
-    GetOverlappedResultEx.restype = wintypes.BOOL
+    # Windows 7 and earlier does not support GetOverlappedResultEx
+    # The alternative is to use GetOverlappedResult and wait
+    # for read or write operation to complete. This is
+    # done be using CreateEvent and WaitForSingleObject.
+    # Windows 8 is version 6.2 - see the Operating System Version
+    # page on MSDN
+    WINDOWS_8_AND_LATER = sys.getwindowsversion() >= (6, 2)
```

Comment:
It would be great to exercise both variations on this code on the CI system.  We have some infrastructure for this sort of thing.  Take a look in `tests/integration/WatchmanTestCase.py` for the `expandConfigurations` method.

For windows I think we want to introduce a dimension there (Something like `('local', 'windows-7-bser')`) that forces us to use the Windows 7 implementation for one set of running the tests.